File: examples/vulnerable_example.py
"""
examples/vulnerable_example.py — intentionally insecure code for demo purposes.

Run:  codeguard scan examples/vulnerable_example.py
"""

import pickle
import sqlite3
import subprocess

# CG-SEC-002: hardcoded secret
api_key = "sk-super-secret-12345"


def get_user(username: str) -> dict:
    conn = sqlite3.connect("users.db")
    cursor = conn.cursor()
    # CG-SEC-001: SQL injection via f-string
    cursor.execute(f"SELECT * FROM users WHERE username = '{username}'")
    return cursor.fetchone()


def run_report(report_name: str) -> str:
    # CG-SEC-005: shell=True with dynamic argument
    result = subprocess.run(
        f"generate_report.sh {report_name}", shell=True, capture_output=True, text=True
    )
    return result.stdout


def load_cached_model(data: bytes) -> object:
    # CG-SEC-004: unsafe deserialization
    return pickle.loads(data)


def evaluate(expression: str) -> object:
    # CG-SEC-003: eval on dynamic input
    return eval(expression)

File: examples/vulnerable_example.js
/*
 * examples/vulnerable_example.js -- intentionally insecure code for demo purposes.
 *
 * Run:  codeguard scan examples/vulnerable_example.js
 */

const cp = require("child_process");

// CG-SEC-105: hardcoded secret
const apiKey = "sk-live-super-secret-12345";

function runReport(userInput, branch) {
  // CG-SEC-101: eval on user input
  const parsed = eval(userInput);

  // CG-SEC-102: shell command injection
  cp.execSync("git checkout " + branch);

  // CG-SEC-106: Math.random() for a security token
  const sessionToken = Math.random().toString(36).slice(2);

  return { parsed, sessionToken, apiKey };
}

function render(el, userHtml) {
  // CG-SEC-103: DOM XSS sink
  el.innerHTML = userHtml;
}

module.exports = { runReport, render };
